### PR TITLE
FW-4761 Add import for dictionary related entries and media

### PR DESCRIPTION
**Pull request by bronson-c**
### Description of Changes
Adds import for related dictionary entries and media.

### Checklist
- [ ] README / documentation has been updated if needed
- [ ] PR title / commit messages contain Jira ticket number if applicable
- [ ] Tests have been updated / created if applicable
- [ ] Admin Panel has been updated if models have been added or modified
- [ ] Migrations have been updated and committed if applicable
- [ ] Team Postman workspace has been updated if applicable

### Reviewers Should Do The Following:
- [ ] Review the code changes here
- [ ] Pull the branch and test locally

### Additional Notes
Apologies for the wrong ticket number in the branch name, must've been distracted 😅 

TODO: Skip rows with missing categories and missing related entries.

### CODE REVIEW

File: firstvoices/backend/resources/dictionary.py [hunk 1]
```diff
@@ -92,17 +112,17 @@ class DictionaryEntryCategoryResource(BaseResource):
...
         try:
             Category.objects.get(id=row["category"])
         except Category.DoesNotExist:
             logger.warning(
-                f"Skipping row with category id {row['category']} because it does not exist"
+                f"Skipping row {row_number} because entry {row['category']} does not exist."
             )
-            raise self.skip_row("Category does not exist")
+            row["category"] = ""
+            # TODO: prevent the row from being imported
```
Comment (bronson-c) [inline firstvoices/backend/resources/dictionary.py:125]: I could not quite figure out how to skip importing the row, if someone wants to pick this ticket up this is the only work left to do. Thanks!

File: firstvoices/backend/resources/dictionary.py [hunk 2]
```diff
@@ -119,3 +139,32 @@ class DictionaryEntryRelatedCharacterResource(BaseResource):
...
+        # skip rows with non-existent "to" and "from" dictionary entries
+        logger = logging.getLogger(__name__)
+        try:
+            DictionaryEntry.objects.get(id=row["related_entry"])
+        except DictionaryEntry.DoesNotExist:
+            logger.warning(
+                f"Skipping row {row_number} because entry {row['related_entry']} does not exist."
+            )
+            row["related_entry"] = ""
+            # TODO: prevent the row from being imported
```
Comment (bronson-c) [inline firstvoices/backend/resources/dictionary.py:170]: Same as above comment: I could not quite figure out how to skip importing the row, if someone wants to pick this ticket up this is the only work left to do. Thanks!

**Review by NolanVH — APPROVED** [review]
All changes look good to me! Thanks!

File: firstvoices/backend/tests/test_resources/test_dictionary_resources.py [hunk 3]
```diff
@@ -194,21 +196,27 @@ def test_import_base_data_with_nonexistent_category(self):
...
+        category2 = factories.CategoryFactory.create(site=site)
         data = [
-            f"{uuid.uuid4()},,,,,{site.id},{dictionary_entry.id},nonexistent_category",
+            f"{uuid.uuid4()},,,,,{site.id},{dictionary_entry.id},{category.id}",
+            f"{uuid.uuid4()},,,,,{site.id},{dictionary_entry.id},{uuid.uuid4()}",  # non-existent category
+            f"{uuid.uuid4()},,,,,{site.id},{dictionary_entry.id},{category2.id}",
         ]
 
         table = self.build_table(data)
         result = DictionaryEntryCategoryResource().import_data(dataset=table)
```
Comment (caforbes) [inline firstvoices/backend/tests/test_resources/test_dictionary_resources.py:208]: Does this still work if the import is set to `raise_errors=True` as it is in the script? The tests are not set to raise errors and halt during import but the actual script is, and I wonder if that would present issues with this way of catching nonexistent entries.

**Review by caforbes — APPROVED** [review]
If the skipping works for imports where `raise_errors=True`, then looks good! If not, then we need to choose between running the actual import script with `raise_errors=False`, or reworking how rows are skipped (e.g. add a try/except in `skip_row` or something).

File: firstvoices/backend/tests/test_resources/test_dictionary_resources.py (hunk 3 above)
Comment (sahil97) [reply firstvoices/backend/tests/test_resources/test_dictionary_resources.py:208]: Yes, so I just tested with both raise_errors=True in the test cases, and also manually editing categories to have wrong UUIDs in the csv file, and while importing the incorrect csv file, the script does not halt the import process and marks the row as skipped.
So merging the PR as of now. Please let me know if any additional changes need to be made.

**Pull request merged**